src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpTraceHelper.cs

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Security;
+
+namespace System.Net.Http
+{
+    internal static class WinHttpTraceHelper
```

Comment:
Also discussed offline: we should switch to Log/GlobalLog ETW-based tracing for all our components to allow this to work for non-debug builds as well.